zambeze/orchestration/processor.py

```diff
@@ -189,59 +203,60 @@ async def __process_files(self, files: list[str]) -> None:
-                    #                     "type": "synchronous",
-                    #                     "items": [file_url],
-                    #                 }
-                    #             },
-                    #         ],
-                    #     },
-                    # )
-                    raise Exception("Needs to be implemented.")
+                ]
+                # Will validate the message fields and then make it immutable
+                immutable_msg_move = self._msg_factory.create(msg_template_move)
+                checked_result = self._settings.plugins.check(immutable_msg_move)
+                self._logger.debug(checked_result)
+
+                await self._queue_client.send(ChannelType.ACTIVITY, immutable_msg)
+                await self._queue_client.send(ChannelType.ACTIVITY, immutable_msg_move)
+
+                # If the local agent does not support Globus we will need to
+                # send a request to to nats for someone else to handle the
+                # transfer
+                # await self.send(
+                #     MessageType.COMPUTE.value,
+                #     {
+                #         "plugin": "globus",
+                #         "cmd": [
+                #             {
+                #                 "transfer": {
+                #                     "type": "synchronous",
+                #                     "items": [file_url],
+                #                 }
+                #             },
+                #         ],
+                #     },
+                # )
+                # raise Exception("Needs to be implemented.")
 
             elif file_url.scheme == "rsync":
-                await self._queue_client.send(
-                    ChannelType.ACTIVITY,
-                    {
-                        "plugin": "rsync",
-                        "cmd": [
-                            {
-                                "transfer": {
-                                    "source": {
-                                        "ip": file_url.netloc,
-                                        "path": file_url.path,
-                                        "user": file_url.username,
-                                    },
-                                    "destination": {
-                                        "ip": socket.gethostbyname(
-                                            socket.gethostname()
-                                        ),
-                                        "path": str(pathlib.Path().resolve()),
-                                        "user": getpass.getuser(),
-                                    },
-                                }
-                            }
-                        ],
-                    },
+
+                msg_template = self._msg_factory.createTemplate(
+                    MessageType.ACTIVITY, "rsync", "transfer"
                 )
 
+                msg_template[1]["body"]["cmd"] = [
```

Comment:
Repeat comment from earlier (are we editing second element of list? and can this be done cleaner)